jmix-multitenancy/multitenancy/src/main/java/io/jmix/multitenancy/core/impl/TenantProviderImpl.java

```diff
@@ -46,14 +48,33 @@ public String getCurrentUserTenantId() {
         if (!currentAuthentication.isSet()) {
             return TenantProvider.NO_TENANT;
         }
+
         if (!(currentAuthentication.getAuthentication().getPrincipal() instanceof UserDetails)) {
             return TenantProvider.NO_TENANT;
         }
+
         UserDetails userDetails = currentAuthentication.getUser();
-        if (!(userDetails instanceof AcceptsTenant)) {
+        String tenantIdFieldName = getTenantIdFieldName(userDetails.getClass());
+
+        if (tenantIdFieldName == null) {
```

Comment:
Check `AcceptsTenant` presence, just in case

Reply:
As far as I understand from the issue description, we are trying to completely get rid of the use of the `AcceptTenants` interface.

That's why it's marked as `deprecated`.

During migration nothing will change for end-users, since now only the presence of the `@TenantId` annotation above the field is required. The interface check in this case, as it seems to me, will be redundant.